--- a/drivers/video/video-uclass.c
+++ b/drivers/video/video-uclass.c
@@ -579,6 +579,7 @@
 	return priv->ysize;
 }
 
+#if defined(CONFIG_VIDEO_LOGO)
 #define SPLASH_DECL(_name) \
 	extern u8 __splash_ ## _name ## _begin[]; \
 	extern u8 __splash_ ## _name ## _end[]
@@ -586,20 +587,28 @@
 #define SPLASH_START(_name)	__splash_ ## _name ## _begin
 
 SPLASH_DECL(u_boot_logo);
+#endif
 
 void *video_get_u_boot_logo(void)
 {
+#if defined(CONFIG_VIDEO_LOGO)
 	return SPLASH_START(u_boot_logo);
+#else
+	return NULL;
+#endif
 }
 
 static int show_splash(struct udevice *dev)
 {
+	int ret = 0;
+
+#if defined(CONFIG_VIDEO_LOGO)
 	u8 *data = SPLASH_START(u_boot_logo);
-	int ret;
 
 	ret = video_bmp_display(dev, map_to_sysmem(data), -4, 4, true);
+#endif
 
-	return 0;
+	return ret;
 }
 
 int video_default_font_height(struct udevice *dev)
